### ISSUE TITLE
Restore-DbaDatabase throws PercentCompleteEventHandler issues

### ISSUE THREAD

**Issue by potatoqualitee** [sqlcollaborative/dbatools#3471]
got this in an email

```
Hi.

I am working on a script that copies databases from once source to multiple destinations and I am trying out the Copy-DbaDatabase command. I have run into a problem here.
The first time when the command is executed, it works fine. The second time always fails  with following error:

Backup-DbaDatabase : Cannot convert argument "value", with value:
"Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler", for "add_PercentComplete" to type
"Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler": "Cannot convert the
"Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" value of type
"Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" to type
"Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler"."
At C:\Program Files\WindowsPowerShell\Modules\dbatools\allcommands.ps1:6034 char:44
+                         $backupTmpResult = Backup-DbaDatabase -SqlInstance $sour ...
+ ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    + CategoryInfo          : NotSpecified: (:) [Backup-DbaDatabase], MethodException
    + FullyQualifiedErrorId : MethodArgumentConversionInvalidCastArgument,Backup-DbaDatabase

Same input is used both times.
A solution is always to re-start the powershell console and it works fine again.
Have you seen this error before?

I am using Powershell 4.0.
I got around the issue by starting a new powershell session, attaching the command to it.

Example: 

powershell.exe -Command {Copy-DbaDatabase .....}
This way I was able to execute the Copy-DbaDatabase command more than once in a single script.
```

I asked what version of PS he was using, figuring it was 4. Sure enough. I'm sorry I forgot to ask the version of dbatools.

Chrissy

**Comment by potatoqualitee** [comment]
Thanks, yall!

**Comment by niphlod** [comment]
for the life of me I can't replicate. Tried my best but just couldn't.

**Comment by beargle** [comment]
Was able to reproduce in v4 when multiple versions of SMO are loaded:

```
PS> $psversiontable.psversion

Major  Minor  Build  Revision
-----  -----  -----  --------
4      0      -1     -1

PS> [appdomain]::currentdomain.getassemblies() | sort -property fullname | where-object { $_.fullname -like '*Smo,*' } | format-table fullname

FullName
--------
Microsoft.SqlServer.Smo, Version=13.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91
Microsoft.SqlServer.Smo, Version=14.0.0.0, Culture=neutral, PublicKeyToken=89845dcd8080cc91
```

**Can't reproduce w/ dbatools in isolation**. Have only seen it when mixing native SMO and dbatools calls in the same session, and different SMO assembly versions are available (ex. an old SqlServer module version that was installed with SSMS 16.5.3).

**Comment by samuelthan** [comment]
I think i'm in the same boat. However it's triggered using invoke-dbaloggshipping

My environment = SQL Server 2017 Web Edition @ Windows 2016 Server

```
$psversiontable.psversion
Major  Minor  Build  Revision
-----  -----  -----  --------
5      1      14393  2395 
```   



This is the command i used
`Invoke-DbaLogShipping -SourceSqlInstance $sourceServer -DestinationSqlInstance $destinationServer -Database $database -BackupNetworkPath "\\networksharedpath1\logshipping"  -CompressBackup -GenerateFullBackup -CopyDestinationFolder "\\SECONDARYSQL\logshipping" -NoRecovery -Verbose -EnableException -Force`


Sometimes it works, sometimes it doesn't


```
Error output 
PSMessageDetails      :
Exception             : System.Exception: Cannot convert argument "value", with value: "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler", for "add_PercentComplete" to type
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler": "Cannot convert the "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" value of type
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" to type "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler"." --->
                       System.Management.Automation.MethodException: Cannot convert argument "value", with value: "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler", for
                       "add_PercentComplete" to type "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler": "Cannot convert the
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" value of type "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" to type
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler"." ---> System.Management.Automation.PSInvalidCastException: Cannot convert the
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" value of type "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler" to type
                       "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler".
                          at System.Management.Automation.LanguagePrimitives.ThrowInvalidCastException(Object valueToConvert, Type resultType)
                          at System.Management.Automation.LanguagePrimitives.ConvertNoConversion(Object valueToConvert, Type resultType, Boolean recurse, PSObject originalValueToConvert,
                       IFormatProvider formatProvider, TypeTable backupTable)
                          at CallSite.Target(Closure , CallSite , Object , Object )
                          --- End of inner exception stack trace ---
                          at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
                          at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
                          at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                          at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
                          at System.Management.Automation.Interpreter.Interpreter.Run(InterpretedFrame frame)
                          at System.Management.Automation.Interpreter.LightLambda.RunVoid1[T0](T0 arg0)
                          at System.Management.Automation.PSScriptCmdlet.RunClause(Action`1 clause, Object dollarUnderbar, Object inputToProcess)
                          at System.Management.Automation.PSScriptCmdlet.DoProcessRecord()
                          at System.Management.Automation.CommandProcessor.ProcessRecord()
                          --- End of inner exception stack trace ---
TargetObject          : TESTSQL1
CategoryInfo          : NotSpecified: (TESTSQL1:String) [], Exception
FullyQualifiedErrorId : dbatools_Invoke-DbaLogShipping
ErrorDetails          :
InvocationInfo        : System.Management.Automation.InvocationInfo
ScriptStackTrace      : at Stop-Function, C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.387\allcommands.ps1: line 95730
                       at Invoke-DbaLogShipping<Process>, C:\Program Files\WindowsPowerShell\Modules\dbatools\0.9.387\allcommands.ps1: line 52161
```

**Comment by potatoqualitee** [comment]
We recently removed the global variable. Can you test again, @samuelthan and @beargle ? 🙇

**Comment by potatoqualitee** [comment]
Will close unless there's additional feedback

**Comment by GrahamDBA** [comment]
I tried to solve this error when using : Copy-DbaDatabase

> Cannot convert argument "value", with value: "Microsoft.SqlServer.Management.Smo.PercentCompleteEventHandler", for "add_PercentComplete" to ...

First I found man sessions of Windows PowerShell_ISE open so I killed them.
Then I found the error disappeared when I used the most simplest for of the commands.
But it failed when I included some PowerShell functions of my own.
So copied over the essential functions I required and I removed the include line and it worked fine.
